src/app/index.test.js

```diff
@@ -0,0 +1,682 @@
+            return resetGitRepository();
+          });
+
+          it('emits "onNoVersionChanges" event', async () => {
+            expect(spies.onNoVersionChanges).to.have.been.calledWith(SERVICE_A_ID, SERVICE_A_TYPE);
+          });
+
+          it('emits no other events', async () => {
+            expect(spies.onStartTrackingChanges).to.have.not.been.called;
+            expect(spies.onEndTrackingChanges).to.have.not.been.called;
+            expect(spies.onStartRefiltering).to.have.not.been.called;
+            expect(spies.onEndRefiltering).to.have.not.been.called;
+            expect(spies.onFirstSnapshotRecorded).to.have.not.been.called;
+            expect(spies.onSnapshotRecorded).to.have.not.been.called;
+            expect(spies.onFirstVersionRecorded).to.have.not.been.called;
+            expect(spies.onVersionRecorded).to.have.not.been.called;
+            expect(spies.onNoSnapshotChanges).to.have.not.been.called;
+            expect(spies.onChangesPublished).to.have.not.been.called;
+            expect(spies.onApplicationError).to.have.not.been.called;
+            expect(spies.onRecordRefilterError).to.have.not.been.called;
+            expect(spies.onDocumentUpdateError).to.have.not.been.called;
+            expect(spies.onDocumentFetchError).to.have.not.been.called;
+          });
+        });
+      });
+    });
+
+    describe('#fetch', () => {
+      after(() => {
+        resetSpiesHistory();
+        return resetGitRepository();
+      });
+
+      context('When everything is ok', () => {
+        before(async () => {
+          nock('https://www.servicea.example').get('/tos').reply(200, SERVICE_A_TOS_SNAPSHOT);
+
+          const { fetch: location } = app.serviceDeclarations[SERVICE_A_ID].documents[SERVICE_A_TYPE];
+
+          return app.fetch({ location, serviceId: SERVICE_A_ID, type: SERVICE_A_TYPE });
+        });
+
+        it('emits no events', async () => {
+          expect(spies.onStartTrackingChanges).to.have.not.been.called;
+          expect(spies.onEndTrackingChanges).to.have.not.been.called;
+          expect(spies.onStartRefiltering).to.have.not.been.called;
+          expect(spies.onEndRefiltering).to.have.not.been.called;
+          expect(spies.onFirstSnapshotRecorded).to.have.not.been.called;
+          expect(spies.onSnapshotRecorded).to.have.not.been.called;
+          expect(spies.onNoSnapshotChanges).to.have.not.been.called;
+          expect(spies.onFirstVersionRecorded).to.have.not.been.called;
+          expect(spies.onVersionRecorded).to.have.not.been.called;
+          expect(spies.onNoVersionChanges).to.have.not.been.called;
+          expect(spies.onChangesPublished).to.have.not.been.called;
+          expect(spies.onApplicationError).to.have.not.been.called;
+          expect(spies.onRecordRefilterError).to.have.not.been.called;
+          expect(spies.onDocumentUpdateError).to.have.not.been.called;
+          expect(spies.onDocumentFetchError).to.have.not.been.called;
```

Comment:
All these calls look very unmaintainable to me: every time we'll update an event name, create or remove an event, we'll have to update hundreds of lines 😕

I am not 100% sure of the added value of testing that all other events have not been called, but if you want to test it, what about creating a helper test function that takes as a parameter the name of the only event that is expected to have been called, checks that is has been called, and iterates on all the others to check that they haven't? 🙂